src/main/groovy/com/aoe/scmpushtrigger/ScmPushTrigger.groovy

```diff
@@ -69,31 +75,55 @@ class ScmPushTrigger extends Trigger<AbstractProject<?, ?>> {
     }
 
     @Override
-    void start(AbstractProject<?, ?> project, boolean newInstance) {
-        super.start(project, newInstance)
+    void start(Job job, boolean newInstance) {
+        super.start(job, newInstance)
 
         PushNotificationProviderAccess.instance.addTrigger(getTriggerRef())
     }
 
     @Override
     void stop() {
-        PushNotificationProviderAccess.instance.removeTrigger(getTriggerRef())
+        //while stop is called, we don't need creation of a new trigger ref
+        if (triggerRef) {
+            PushNotificationProviderAccess.instance.removeTrigger(getTriggerRef())
+        }
         super.stop()
     }
 
     boolean matches(String content) {
         if (matchExpression) {
             matchExpression == content
         } else {
-            getScmUrls().contains(content)
+            def scmUrls        = getScmUrls()
+            def matchResult    = scmUrls.contains(content)
+
+            if(matchResult) {
+                matchResult
+
+            } else {
+
+                //try to compare scm urls with content, that has .git suffix removed
+                //this is for jobs with configured git-urls without .git suffix
+                //the message (content) from rabbit mq always contains .git
+                def dotGitSuffix = '.git'
+                def trimmedContent = content.trim()
```

Comment:
Is that necessary? If so: Shouldn't it be moved above the `if` and also used in the first comparison?